jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/datagrip/actions/SecretsManagerDialogWrapper.kt

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.datagrip.actions
+
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.progress.PerformInBackgroundOption
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task.Backgroundable
+import com.intellij.openapi.ui.DialogWrapper
+import software.amazon.awssdk.services.secretsmanager.model.SecretListEntry
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerNode
+import software.aws.toolkits.jetbrains.datagrip.DatabaseSecret
+import software.aws.toolkits.jetbrains.datagrip.auth.SecretsManagerDbSecret
+import software.aws.toolkits.jetbrains.services.secretsmanager.SecretsManagerResources
+import software.aws.toolkits.jetbrains.services.secretsmanager.arnToName
+import software.aws.toolkits.jetbrains.ui.ResourceSelector
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+import java.awt.BorderLayout
+import javax.swing.JComponent
+import javax.swing.JPanel
+
+class SecretsManagerDialogWrapper(private val selected: AwsExplorerNode<*>) : DialogWrapper(selected.nodeProject) {
+    private lateinit var secrets: ResourceSelector<SecretListEntry>
+    var dbSecret: SecretsManagerDbSecret? = null
+        private set
+    var dbSecretArn: String? = null
+        private set
+
+    init {
+        title = message("datagrip.secretsmanager.action.title")
+        setOKButtonText(message("general.create_button"))
+        init()
+    }
+
+    override fun createCenterPanel(): JComponent? {
+        secrets = ResourceSelector.builder(selected.nodeProject)
+            .resource(SecretsManagerResources.secrets)
+            .customRenderer { entry, renderer -> renderer.append(entry.name()); renderer }
+            .build().also {
+                // When it is changed, make sure the OK button is re-enabled
+                it.addActionListener {
+                    isOKActionEnabled = true
+                }
+            }
+        val panel = JPanel(BorderLayout())
+        panel.add(secrets)
+        return panel
+    }
+
+    override fun doOKAction() {
+        if (!okAction.isEnabled) {
+            return
+        }
+        object : Backgroundable(
+            selected.nodeProject,
+            message("datagrip.secretsmanager.validating"),
+            false,
+            PerformInBackgroundOption.ALWAYS_BACKGROUND
+        ) {
+            override fun run(indicator: ProgressIndicator) {
+                try {
+                    validateConfiguration()
+                } catch (e: Exception) {
+                    notifyError(content = e.message ?: "")
```

Comment:
Should we have a title? And a better fallback error

Reply:
Yep, changed